File: BackEnd/backend/src/main/java/com/fitapp/backend/parameter/application/dto/request/CustomParameterFilterRequest.java
package com.fitapp.backend.parameter.application.dto.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fitapp.backend.parameter.infrastructure.persistence.entity.ParameterType;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import org.springframework.data.domain.Sort;

import java.util.ArrayList;
import java.util.List;

@Data
@Schema(description = "Filtros para búsqueda de parámetros personalizados")
public class CustomParameterFilterRequest {
    
    @Schema(description = "Texto de búsqueda en nombre o descripción")
    @JsonProperty("search")
    private String search;
    
    @Schema(description = "Tipo de parámetro")
    @JsonProperty("parameterType")
    private ParameterType parameterType;

    @Schema(description = "Favoritos")
    @JsonProperty("isFavorite")
    private Boolean isFavorite;

    @Schema(description = "Si es global")
    @JsonProperty("isGlobal")
    private Boolean isGlobal;
    
    @Schema(description = "Si está activo")
    @JsonProperty("isActive")
    private Boolean isActive;
    
    @Schema(description = "Si es trackeable para métricas")
    @JsonProperty("isTrackable")
    private Boolean isTrackable;
    
    @Schema(description = "ID del dueño")
    @JsonProperty("ownerId")
    private Long ownerId;
    
    @Schema(description = "Solo mis parámetros")
    @JsonProperty("onlyMine")
    private Boolean onlyMine = false;
    
    @Schema(description = "Página solicitada (0-index)", example = "0")
    @JsonProperty("page")
    private Integer page = 0;
    
    @Schema(description = "Tamaño de página", example = "20")
    @JsonProperty("size")
    private Integer size = 20;
    
    @Schema(description = "Campo para ordenar", example = "name")
    @JsonProperty("sortBy")
    private String sortBy = "name";
    
    @Schema(description = "Dirección del orden", example = "ASC")
    @JsonProperty("direction")
    private Sort.Direction direction = Sort.Direction.ASC;
    
    @Schema(description = "Múltiples campos para ordenar")
    @JsonProperty("sortFields")
    private List<SortField> sortFields = new ArrayList<>();
    
    @Data
    public static class SortField {
        private String field;
        private Sort.Direction direction = Sort.Direction.ASC;
    }
}

File: FrontEnd/app/src/main/java/com/fitapp/appfit/feature/parameter/model/request/CustomParameterFilterRequest.kt
package com.fitapp.appfit.feature.parameter.model.request

import com.google.gson.annotations.SerializedName

data class CustomParameterFilterRequest(
    @SerializedName("search") val search: String? = null,
    @SerializedName("parameterType") val parameterType: String? = null,
    @SerializedName("isGlobal") val isGlobal: Boolean? = null,
    @SerializedName("isActive") val isActive: Boolean? = null,
    @SerializedName("isFavorite") val isFavorite: Boolean? = null,
    @SerializedName("isTrackable") val isTrackable: Boolean? = null,
    @SerializedName("sportId") val sportId: Long? = null,
    @SerializedName("ownerId") val ownerId: Long? = null,
    @SerializedName("category") val category: String? = null,
    @SerializedName("onlyMine") val onlyMine: Boolean = false,
    @SerializedName("includeGlobal") val includeGlobal: Boolean = true,
    @SerializedName("page") val page: Int = 0,
    @SerializedName("size") val size: Int = 20,
    @SerializedName("sortBy") val sortBy: String = "name",
    @SerializedName("direction") val direction: String = "ASC"
)